docs/source/developer_guide/deprecation.rst

```diff
@@ -4,71 +4,78 @@
+A release number denotes <major>.<minor>.<patch> versions.
 
-Our current deprecation process is as follows:
+Our current deprecation policy is that we remove functionality after one minor release cycle.
+For example, if some functionality has been deprecated in v0.9.0 or any patch of that minor version, we will raise a warning throughout the next minor version v0.10.* and all of its patches.
+The functionality will then be removed in v0.11.0.
 
-* We raise a `FutureWarning <https://docs.python.org/3/library/exceptions.html#FutureWarning>`_. The warning message should the give the version number when the functionality will be removed and describe the new usage.
+Deprecation process
+===================
 
-* We add a to-do comments to the lines of code that can be removed, with the version number when the code can be removed. For example, :code:`TODO: remove in v0.10.0`.
+The deprecation process is as follows:
 
-* We remove all deprecated functionality as part of the release process, searching for the to-do comments.
+* **Raise a warning.** For all deprecated functionality, we raise a :code:`FutureWarning`. The warning message should give the version number when the functionality will be changed and describe the new usage.
+* **Add a TODO comment in the code.** Add the following TODO comment to all pieces of code that can be removed: :code:`TODO: remove in <version-number>`. For example, :code:`TODO: remove in v0.11.0`. The TODO comment should describe which steps need to be taken for deprecation (e.g. removal of arguments, removal of functions or blocks of code). If changes need to be applied across multiple places, place multiple TODO comments.
+* **Remove deprecated functionality.** As part of the release process, all deprecated functionality that is due to be removed will be removed by searching for the TODO comments.
 
-We use the `deprecated <https://deprecated.readthedocs.io/en/latest/index.html>`_ package for depreciation helper functions.
 
-To deprecate functionality, we use the :code:`deprecated` decorator.
+To raise the warning, we use the `deprecated <https://deprecated.readthedocs.io/en/latest/index.html>`_ package.
+The package provides depreciation helper functions such as the :code:`deprecated` decorator.
 When importing it from :code:`deprecated.sphinx`, it automatically adds a deprecation message to the docstring.
-You can deprecate functions, methods or classes.
+You can decorate functions, methods or classes.
 
 Examples
-========
+--------
 
-In the examples below, the :code:`deprecated` decorator will raise a FutureWarning saying that the functionality has been deprecated since version 0.8.0 and will be remove in version 0.10.0.
+In the examples below, the :code:`deprecated` decorator will raise a :code:`FutureWarning` saying that the functionality has been deprecated since version 0.8.0 and will be removed in version 0.10.0.
```

Comment:
This statement does not look correct - in the examples, deprecation is since version 0.9.0 and to be removed in 0.11.0, no?